gcloud/vision/client.py

```diff
@@ -0,0 +1,136 @@
+
+
+class Client(JSONClient):
+    """Client to bundle configuration needed for API requests.
+
+    :type project: str
+    :param project: the project which the client acts on behalf of.
+                    If not passed, falls back to the default inferred
+                    from the environment.
+
+    :type credentials: :class:`oauth2client.client.OAuth2Credentials` or
+                       :class:`NoneType`
+    :param credentials: The OAuth2 Credentials to use for the connection
+                        owned by this client. If not passed (and if no ``http``
+                        object is passed), falls back to the default inferred
+                        from the environment.
+
+    :type http: :class:`httplib2.Http` or class that defines ``request()``.
+    :param http: An optional HTTP object to make requests. If not passed, an
+                 ``http`` object is created that is bound to the
+                 ``credentials`` for the current object.
+
+    Usage:
+    >>> from gcloud import vision
+    >>> vision_client = vision.Client()
+    >>> f = open('/tmp/car.jpg', 'r')
+    >>> vision_client.annotate(f.read(), "LABEL_DETECTION", 3)
+
+    Multiple images example:
+    >>> images = (("./image.jpg", ["LABEL_DETECTION", "LANDMARK_DETECTION"]),
+    >>>           ("./image2.jpg", ["FACE_DETECTION", "TEXT_DETECTION"]),)
+    >>> annotated_images = []
+    >>> for image, feature_types in images:
+    >>>     annotated_images.append(vision_client.annotate(image,
+    >>>                                                    feature_types))
+    """
+
+    _connection_class = Connection
+
+    def annotate(self, image, feature_type, max_results=1):
+        """Annotate an image to discover it's attributes.
+
+        :type image: str
+        :param image: A string which can be a URL, a Google Cloud Storage path,
+                      or a byte stream of the image.
+
+        :type feature_type: str or list
+        :param feature_type: The type of detection that the Vision API should
+                             use to determine image attributes. *Pricing is
+                             based on the number of Feature Types.
+
+                             See:
+                             https://cloud.google.com/vision/docs/pricing
+
+
+        :type max_results: int
+        :param max_results: The number of results per feature type to be
+                            returned.
+        """
+        vision_requests = []
+        features = []
+
+        if type(image) == str:
+            img = Image(image)
+
+            if type(feature_type) == list:
+                for feature in feature_type:
+                    features.append(Feature(feature, max_results))
+            else:
+                features.append(Feature(feature_type, max_results))
+
+            vision_requests.append(VisionRequest(img, features))
+
+        data = {"requests": vision_requests}
+        data = json.dumps(data, cls=VisionJSONEncoder)
```

Comment:
Could've you just do:

```python
data = {'request': [req.as_dict() for req in vision_requests]}
```

or even better, don't build up `vision_requests` when you only care about building up a list of

```python
VisionRequest(img, features).as_dict()
```